### DISCUSSION TITLE
add sysroot set

### DISCUSSION THREAD

**Discussion by wangzhankun** [xmake-io/xmake#4370] [Q&A]
### Is your feature request related to a problem? Please describe.

No.

### Describe the solution you'd like

Cross Compile.
When I exec:
```sh
xmake f -p cross --sdk=/opt/riscv-gnu-toolchain -a riscv64 -cvD
configure
{
    arch = riscv64
    plat = cross
    ndk_stdcxx = true
    mode = release
    ccache = true
    clean = true
    kind = static
    target_os = linux
    buildir = build
    host = linux
    sdk = /opt/riscv-gnu-toolchain
}
```
There is no sysroot, but I need to build it with "sys/socket.h".

**Comment by waruqi** [comment]
you can pass it to `--ldflags`, `--cxflags`

**Comment by wangzhankun** [comment]
I only want to change the sysroot when the toolchain is riscv, is there any way to check the toolchain?
Anyway, I hope xmake can support the feature for that there may be a lot third party libs in the sysroot.

**Comment by waruqi** [comment]
> I only want to change the sysroot when the toolchain is riscv, is there any way to check the toolchain? Anyway, I hope xmake can support the feature for that there may be a lot third party libs in the sysroot.

you can define custom toolchain and add ldflags/cxflags in toolchain()
